nemo/collections/asr/data/audio_to_audio.py

```diff
@@ -0,0 +1,936 @@
+        audio_to_manifest_key: Dictionary mapping audio signal labels to manifest keys.
+        audio_duration: Optional duration of each item returned by __getitem__.
+                        If `None`, complete audio will be loaded.
+                        If set, a subsegment will be loaded synchronously from
+                        target and audio, i.e., with the same start and end point.
+        random_offset: If `True`, offset will be randomized when loading a subsegment
+                       from a file.
+        max_duration: If audio exceeds this length, do not include in dataset.
+        min_duration: If audio is less than this length, do not include in dataset.
+        max_utts: Limit number of utterances.
+    """
+
+    @property
+    @abc.abstractmethod
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports.
+        """
+
+    def __init__(
+        self,
+        manifest_filepath: str,
+        sample_rate: int,
+        audio_to_manifest_key: Dict[str, Union[str, List[str]]],
+        audio_duration: Optional[float] = None,
+        random_offset: bool = False,
+        max_duration: Optional[float] = None,
+        min_duration: Optional[float] = None,
+        max_utts: int = 0,
+    ):
+        """Instantiates an audio_dataset.
+        """
+        super().__init__()
+
+        if type(manifest_filepath) == str:
+            manifest_filepath = manifest_filepath.split(',')
+
+        for audio_key, manifest_key in audio_to_manifest_key.items():
+            if type(manifest_key) == str and ',' in manifest_key:
+                audio_to_manifest_key[audio_key] = manifest_key.split(',')
+
+        self.collection = AudioCollection(
+            manifest_files=manifest_filepath,
+            audio_to_manifest_key=audio_to_manifest_key,
+            min_duration=min_duration,
+            max_duration=max_duration,
+            max_number=max_utts,
+        )
+
+        self.sample_rate = sample_rate
+        self.audio_duration = audio_duration
+        self.random_offset = random_offset
+
+    def num_channels(self, signal_key) -> int:
+        """Returns the number of channels for a particular signal in
+        items prepared by this dictionary.
+
+        More specifically, this will get the tensor from the first
+        item in the dataset, check if it's a one- or two-dimensional
+        tensor, and return the number of channels based on the size
+        of the second axis (shape[1]).
+
+        NOTE:
+        This assumes that all examples returned by `__getitem__`
+        have the same number of channels.
+        
+        Args:
+            signal_key: string, used to select a signal from the dictionary
+                        output by __getitem__
+
+        Returns:
+            Number of channels for the selected signal.
+        """
+        # Assumption: whole dataset has the same number of channels
+        item = self.__getitem__(0)
```

Comment:
Do not assume it is an indexed dataset. This may be invalid for tarred datasets.